test/json_test.cpp

```diff
@@ -0,0 +1,248 @@
+#include <migraphx/shape.hpp>
+#include <migraphx/argument.hpp>
+#include <migraphx/literal.hpp>
+#include <migraphx/serialize.hpp>
+#include <migraphx/functional.hpp>
+#include <migraphx/json.hpp>
+#include <test.hpp>
+
+TEST_CASE(empty_value)
+{
+    migraphx::value v;
+    auto json_str = migraphx::to_json_string(v);
+    EXPECT(json_str == "null");
+}
+
+TEST_CASE(empty_value_rev)
+{
+    std::string json_str = "null";
+    migraphx::value v    = migraphx::from_json_string(json_str);
+    migraphx::value ev   = migraphx::value();
+    EXPECT(v == ev);
+}
+
+TEST_CASE(empty_array)
```

Comment:
This test case should probably be called `null_array` since its an array of nulls and not an empty array.